src/runtime/objmodel.cpp

```diff
@@ -3366,8 +3387,17 @@ void rearrangeArguments(ParamReceiveSpec paramspec, const ParamNames* param_name
     for (int i = 0; i < paramspec.num_args - paramspec.num_defaults; i++) {
         if (params_filled[i])
             continue;
-        raiseExcHelper(TypeError, "%s() takes exactly %d arguments (%ld given)", func_name, paramspec.num_args,
-                       argspec.num_args + argspec.num_keywords + varargs.size());
+        long given = argspec.num_args + argspec.num_keywords + varargs.size();
+        int takes = paramspec.num_args;
+        if (!(paramspec.takes_varargs || paramspec.takes_kwargs)) {
+            raiseExcHelper(TypeError, "%s() takes exactly %d argument%s (%ld given)", func_name, takes,
+                           takes != 1 ? "s" : "", given);
+        } else {
+            if (given < takes) {
+                raiseExcHelper(TypeError, "%s() takes at least %d argument%s (%ld given)", func_name, takes,
+                               takes != 1 ? "s" : "", given);
+            }
+        }
```

Comment:
What's supposed to happen if it falls through here and doesn't throw an exception?